File: mobile/GameX/src/com/mycompany/entities/Product.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.entities;

/**
 *
 * @author HP
 */
public class Product {
    private int id;
    private String ProductName;
    private String Description;
    private String Reference;
    private String image;
    private int Stock;
    private int Price;
    int Category_id;

    public int getCategory_id() {
        return Category_id;
    }

    public void setCategory_id(int Category_id) {
        this.Category_id = Category_id;
    }
    

    public Product() {
    }

    public Product(String ProductName, String Description, String Reference, String image, int Stock, int Price) {
        this.ProductName = ProductName;
        this.Description = Description;
        this.Reference = Reference;
        this.image = image;
        this.Stock = Stock;
        this.Price = Price;
    }

    

    public Product(String text) {
    }
     public Product(int id) {
         this.id = id;
    }

    public Product(String Reference, String ProductName, String Description, int Stock, int Price,int Category_id) {
        this.Reference = Reference;
        this.ProductName = ProductName;
        this.Description = Description;
        this.Stock = Stock;
        this.Price = Price;
         this.Category_id = Category_id;
    }


   

  
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getProductName() {
        return ProductName;
    }

    public void setProductName(String ProductName) {
        this.ProductName = ProductName;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String Description) {
        this.Description = Description;
    }

    public String getReference() {
        return Reference;
    }

    public void setReference(String Reference) {
        this.Reference = Reference;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getStock() {
        return Stock;
    }

    public void setStock(int Stock) {
        this.Stock = Stock;
    }

    public int getPrice() {
        return Price;
    }

    public void setPrice(int Price) {
        this.Price = Price;
    }

    @Override
    public String toString() {
        return "Categorie{" + "id=" + id + ", ProductName=" + ProductName + ", Description=" + Description + ", Reference=" + Reference + ", image=" + image + ", Stock=" + Stock + ", Price=" + Price + '}';
    }

 
    

     
}

File: src/Entity/Product.php
<?php

namespace App\Entity;

use App\Repository\ProductRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Symfony\Component\Serializer\Annotation\Groups;




/**
 * @ORM\Entity(repositoryClass=ProductRepository::class)
 * @UniqueEntity("Reference")
 * 
 */
class Product
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")
     * 
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Nom requise")
     * @Groups("post:read")
     * 
     */
    private $ProductName;

    /**
     * @ORM\Column(type="integer")
     * @Assert\Positive
     * @Assert\NotBlank(message="Prix requise")
     * @Groups("post:read")
     */
    private $Price;

    

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Description requise")
     * @Groups("post:read")
     */
    private $Description;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="Stock requise")
     * @Groups("post:read")
     */
    private $Stock;

    /**
     * @ORM\ManyToOne(targetEntity=Category::class, inversedBy="products")
     * @Groups("post:read")
     */
    private $Category;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=255, nullable=false)
     * @Groups("post:read")
     * 
     */
    private $image;

    /**
     * @ORM\Column(type="string", length=255, unique=true)
     * @Assert\NotBlank(message="Reference requise")
     * @Groups("post:read")
     * 
     */
    private $Reference;

    

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getProductName(): ?string
    {
        return $this->ProductName;
    }

    public function setProductName(string $ProductName): self
    {
        $this->ProductName = $ProductName;

        return $this;
    }

    public function getPrice(): ?int
    {
        return $this->Price;
    }

    public function setPrice(int $Price): self
    {
        $this->Price = $Price;

        return $this;
    }



    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): self
    {
        $this->Description = $Description;

        return $this;
    }

    public function getStock(): ?int
    {
        return $this->Stock;
    }

    public function setStock(int $Stock): self
    {
        $this->Stock = $Stock;

        return $this;
    }
    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getCategory(): ?Category
    {
        return $this->Category;
    }

    public function setCategory(?Category $Category): self
    {
        $this->Category = $Category;

        return $this;
    }

    public function getReference(): ?string
    {
        return $this->Reference;
    }

    public function setReference(string $Reference): self
    {
        $this->Reference = $Reference;

        return $this;
    }

    

    

   
}
